ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/NewApiTest.java

```diff
@@ -0,0 +1,351 @@
+      .add("VIEWTIME").add("USERID").add("PAGEID");
+  private JsonArray expectedColumnTypes = new JsonArray().add("BIGINT").add("BIGINT")
+      .add("BIGINT").add("STRING").add("STRING");
+
+  @Test
+  public void shouldExecutePushQueryWithLimit() throws Exception {
+
+    // Given:
+    String sql = "SELECT * from " + PAGE_VIEW_STREAM + " EMIT CHANGES LIMIT " + 2 + ";";
+
+    // When:
+    QueryResponse response = executePushQuery(sql);
+
+    // Then:
+    assertThat(response.rows, hasSize(2));
+    assertThat(response.responseObject.getJsonArray("columnNames"), is(expectedColumnNames));
+    assertThat(response.responseObject.getJsonArray("columnTypes"), is(expectedColumnTypes));
+    assertThat(response.responseObject.getString("queryId"), is(notNullValue()));
+  }
+
+  @Test
+  public void shouldFailWithInvalidSql() throws Exception {
+
+    // Given:
+    String sql = "SLECTT * from " + PAGE_VIEW_STREAM + " EMIT CHANGES;";
+
+    // Then:
+    shouldFail(sql, "line 1:1: mismatched input 'SLECTT' expecting");
+  }
+
+  @Test
+  public void shouldFailWithMoreThanOneStatement() throws Exception {
+
+    // Given:
+    String sql = "SELECT * from " + PAGE_VIEW_STREAM + " EMIT CHANGES;" +
+        "SELECT * from " + PAGE_VIEW_STREAM + " EMIT CHANGES;";
+
+    // Then:
+    shouldFail(sql, "Expected exactly one KSQL statement; found 2 instead");
+  }
+
+  @Test
+  public void shouldFailWithNonQuery() throws Exception {
+
+    // Given:
+    String sql =
+        "CREATE STREAM SOME_STREAM AS SELECT * from " + PAGE_VIEW_STREAM + " EMIT CHANGES;";
+
+    // Then:
+    shouldFail(sql, "Not a query");
+  }
+
+  @Test
+  public void shouldExecutePushQueryNoLimit() throws Exception {
+
+    KsqlEngine engine = (KsqlEngine) REST_APP.getEngine();
+    // One persistent query for the agg table
+    assertThatEventually(engine::numberOfLiveQueries, is(1));
+
+    // Given:
+    String sql = "SELECT * from " + PAGE_VIEW_STREAM + " EMIT CHANGES;";
+
+    // Create a write stream to capture the incomplete response
+    ReceiveStream writeStream = new ReceiveStream(vertx);
+
+    // Make the request to stream a query
+    JsonObject properties = new JsonObject();
+    JsonObject requestBody = new JsonObject()
+        .put("sql", sql).put("properties", properties);
+    VertxCompletableFuture<HttpResponse<Void>> responseFuture = new VertxCompletableFuture<>();
+    client.post(8089, "localhost", "/query-stream")
+        .as(BodyCodec.pipe(writeStream))
+        .sendJsonObject(requestBody, responseFuture);
+
+    assertThatEventually(engine::numberOfLiveQueries, is(2));
+
+    // Wait for all rows in the response to arrive
+    assertThatEventually(() -> {
+      try {
+        Buffer buff = writeStream.getBody();
+        QueryResponse queryResponse = new QueryResponse(buff.toString());
+        return queryResponse.rows.size();
+      } catch (Throwable t) {
+        return Integer.MAX_VALUE;
```

Comment:
is there any way to communicate this error so that if the test fails here we can figure out why?